### PR TITLE
add driver for ONFI NAND Flash

### PR DESCRIPTION
**Pull request by rogeryou**
Add ONFI NAND Flash driver for using ONFI NAND Flash like MX30LF2GE8AB. 
This driver is tested on MIMXRT1170_EVK.
The implementation is mainly inspired from NXP SDK.
And the onfi.h file unify the interface to access the ONFI NAND Flash controller.
As for controller layer, we add the SEMC module related driver to support the NAND Flash operation.
By the way, NAND Flash MX30LF2GE8AB's on-die ECC is always enable. 
But the implementation also provide software ECC algorithm like BCH.

### CODE REVIEW

File: boards/arm/mimxrt1170_evk/mimxrt1170_evk.dtsi.orig [hunk 1]
```diff
@@ -0,0 +1,149 @@
+/*
```
Comment (desowin) [inline boards/arm/mimxrt1170_evk/mimxrt1170_evk.dtsi.orig:1]: I believe the inclusion of this file is not intended.

File: boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm7.dts.orig [hunk 2]
```diff
@@ -0,0 +1,210 @@
+/*
```
Comment (desowin) [inline boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm7.dts.orig:1]: I believe the inclusion of this file is not intended.

File: drivers/flash/CMakeLists.txt.orig [hunk 3]
```diff
@@ -0,0 +1,95 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (desowin) [inline drivers/flash/CMakeLists.txt.orig:1]: I believe the inclusion of this file is not intended.

File: drivers/flash/CMakeLists.txt.rej [hunk 4]
```diff
@@ -0,0 +1,11 @@
+--- drivers/flash/CMakeLists.txt
```
Comment (desowin) [inline drivers/flash/CMakeLists.txt.rej:1]: I believe the inclusion of this file is not intended.

File: drivers/flash/Kconfig.mcux.rej [hunk 5]
```diff
@@ -0,0 +1,23 @@
+--- drivers/flash/Kconfig.mcux
```
Comment (desowin) [inline drivers/flash/Kconfig.mcux.rej:1]: I believe the inclusion of this file is not intended.

File: drivers/memc/CMakeLists.txt.orig [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (desowin) [inline drivers/memc/CMakeLists.txt.orig:1]: I believe the inclusion of this file is not intended.

File: drivers/memc/Kconfig.mcux.orig [hunk 7]
```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2020 NXP
```
Comment (desowin) [inline drivers/memc/Kconfig.mcux.orig:1]: I believe the inclusion of this file is not intended.

**Review by desowin — CHANGES_REQUESTED** [review]
I didn't yet look at the actual code, however the commit seems to contain unnecessary files. Also the commit message does not describe the changes and missed Signed-off-by line.

**Comment by carlescufi** [comment]
@rogeryou thank you for your contribution.

Please format your commits according to the contribution guidelines:
https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines

Also, since you've already opened a few PRs that have not been merged, feel free to join us on Discord for additional help:
https://chat.zephyrproject.org/

**Comment by carlescufi** [comment]
@rogeryou please don't close the Pull Request, just update it with:
`git push -f origin macronix_onfi_nand`

If you need more help please ping me on [Discord](https://chat.zephyrproject.org/)

**Comment by rogeryou** [comment]
> @rogeryou please don't close the Pull Request, instead amend the commits and then just update it with: `git push -f origin macronix_onfi_nand`
> 
> If you need more help please ping me on [Discord](https://chat.zephyrproject.org/)

OK!Some unnecessary code appears in the current branch.
 I will update the source code, and reopen this PR.

**Pull request closed without merging**